src/pygama/lgdo/lh5_store.py

```diff
@@ -249,11 +267,9 @@ def read_object(self, name, lh5_file, start_row=0, n_rows=sys.maxsize, idx=None,
             elif isinstance(field_mask, (list, tuple)):
                 field_mask = defaultdict(lambda : False, { field : True for field in field_mask} )
             elif not isinstance(field_mask, defaultdict):
-                print('bad field_mask of type', type(field_mask).__name__)
-                return None, 0
+                raise RuntimeError('bad field_mask of type', type(field_mask).__name__)
         elif field_mask is not None:
-            print(f'Warning: datatype {datatype} does not accept a field_mask')
-
+            raise RuntimeError(f'datatype {datatype} does not accept a field_mask')
```

Comment:
I turned this warning into an exception. Or it does make sense to continue the execution?

Reply:
I agree it should be an exception